tests/integration_tests/test_run.py

```diff
@@ -0,0 +1,336 @@
+
+
+def print_caplog(caplog):
+
+    print("\nmessages")
+    for ii, msg in enumerate(caplog.messages):
+        print(f"{ii:2d} {msg}")
+    print("\nrecords")
+    for ii, rec in enumerate(caplog.records):
+        print(f"{ii:2d} {rec}")
+
+
+def search_caplog(caplog, find_me):
+    """Search caplog message for find_me, return message"""
+
+    for msg in caplog.messages:
+        if find_me in msg:
+            return msg
+    return ""
+
+
+def search_caplog_contains(caplog, find_me, contains_me):
+    """Search caplog message for find_me, return true if it contains contains_me"""
+
+    for msg in caplog.messages:
+        if find_me in msg:
+            print(f"Found '{find_me}' in '{msg}'")
+            if contains_me in msg:
+                print(f"Found '{contains_me}' in '{msg}'")
+                return True
+    return False
+
+
+def print_captured(captured):
+
+    print("\nout")
+    for ii, msg in enumerate(captured.out.split("\n")):
+        print(f"{ii:2d} {msg}")
+    print("\nerr")
+    for ii, msg in enumerate(captured.err.split("\n")):
+        print(f"{ii:2d} {msg}")
+
+
+#
+#  Tests
+#
+
+ANATOMICAL_STR = "anatomical is '/flywheel/v0/input/anatomical/dicoms/1.2.826.0.1.3680043.8.498.81096423295716363709677774784503056177.MR.dcm'"
+
+
+def test_convert_stats_works(caplog):
+
+    caplog.set_level(logging.DEBUG)
+
+    install_gear("stats.zip")
+
+    with open("/tmp/gear_environ.json", "r") as f:
+        environ = json.load(f)
+
+    run.do_gear_convert_stats("sub-TOME3024", False, environ, log)
+
+    print_caplog(caplog)
+
+    assert Path("output/sub-TOME3024_rh_aparc_stats_area_mm2.csv").exists()
+
+
+def test_convert_volumes_works(caplog):
+
+    caplog.set_level(logging.DEBUG)
+
+    install_gear("convert.zip")
+
+    mri_dir = f"{str(subjects_dir)}/sub-TOME3024/mri"
+
+    with open("/tmp/gear_environ.json", "r") as f:
+        environ = json.load(f)
+
+    config = {
+        "gear-hippocampal_subfields": True,
+        "gear-brainstem_structures": True,
+    }
+    run.do_gear_convert_volumes(config, mri_dir, False, environ, log)
+
+    print_caplog(caplog)
+
+    assert Path("output/brainstemSsLabels.v12.FSvoxelSpace.nii.gz").exists()
+
+
+def test_brainstem_works(caplog):
+
+    TestCase.skipTest("", f"Test takes too long.")
+
+    caplog.set_level(logging.DEBUG)
+
+    install_gear("hippo.zip")
+
+    mri_dir = f"{str(subjects_dir)}/sub-TOME3024/mri"
+
+    with open("/tmp/gear_environ.json", "r") as f:
+        environ = json.load(f)
+
+    run.do_gear_brainstem_structures("sub-TOME3024", mri_dir, False, environ, log)
+
+    print_caplog(caplog)
+
+    assert 0
+
+
+def test_hippo_works(caplog):
+
+    TestCase.skipTest("", f"Test takes too long.")
+
+    caplog.set_level(logging.DEBUG)
+
+    install_gear("hippo.zip")
+
+    mri_dir = f"{str(subjects_dir)}/sub-TOME3024/mri"
+
+    with open("/tmp/gear_environ.json", "r") as f:
+        environ = json.load(f)
+
+    # This takes 20 minutes!
```

Comment:
How do we configure our testing such that this is possible somewhere?

Reply:
Nicolas said that a 20 minute test run by CI/CD is fine.  There are two of them here and there will be another.  I run them a time or two locally but mostly just leave them skipped to save time.  They also get run on the platform during the end-to-end test.